include/circt/Dialect/SV/SVExpressions.td

```diff
@@ -209,3 +209,41 @@ def SampledOp : SVOp<"system.sampled", [SameOperandsAndResultType]> {
   let results = (outs AnyType:$sampledValue);
   let assemblyFormat = "$expression attr-dict `:` qualified(type($expression))";
 }
+
+def ReorderOp: SVOp<"reorder", [InferTypeOpInterface, NoSideEffect]> {
```

Comment:
I think you should do a comb operator.  reorder isn't a direct verilog operation, so it probably doesn't belong in this dialect.